srcs/incs/IResponseFactory.hpp

```diff
@@ -0,0 +1,10 @@
+#ifndef SRCS_INCS_IRESPONSEFACTORY_HPP_
```

Comment:
I understand why do we have IResponse (we have several concrete responses inheriting from IResponse, IResponse provides minimum common denominator for all of them as a contract on what a response needs to have), but the IResponseFactory? I don't get it.
We only ever create HttpResponseFactory with this, and the pure method you define here is kind of minimalistic.

Reply:
You are right, this may be a little bit over-engineered. The idea of having a `IResponseFactory` and a `HttpResponseFactory` was to apply, completely, the factory design pattern. Since this project's scope is only to handle http protocol, this interface won't be needed. But, with the idea in mind of making the connection unaware of the protocol, and having a response builder class (`HttpResponseFactory`) including that interface allows having rounded design and completely decoupling it from the http protocol.